compiler/rustc_metadata/src/native_libs.rs

```diff
@@ -433,6 +434,52 @@ impl Collector<'tcx> {
                 }
             }
         };
-        DllImport { name: item.ident.name, ordinal: None, calling_convention, span: item.span }
+
+        DllImport {
+            name: item.ident.name,
+            ordinal: self.extract_ordinal(item),
+            calling_convention,
+            span: item.span,
+        }
+    }
+
+    fn extract_ordinal(&self, foreign_function: &hir::ForeignItemRef<'_>) -> Option<u16> {
+        let mut result: Option<u16> = None;
+
+        for attr in self
+            .tcx
+            .hir()
+            .attrs(foreign_function.id.hir_id())
+            .iter()
+            .filter(|a| a.has_name(sym::link_ordinal))
+        {
+            let items = match attr.meta_item_list() {
```

Comment:
This can be added to the top in the form of a filter_map invocation. `filter_map(|attr| attr.meta_item_list())`. Doesn't have to though, if attr is needed in the loop's body e.g. to emit a misformatting error on its span.

Reply:
obsolete due to refactoring